raft/status.go

```diff
@@ -47,3 +49,20 @@ func getStatus(r *raft) Status {
 
 	return s
 }
+
+func (s Status) MarshalJSON() ([]byte, error) {
```

Comment:
It would also be nice if Status had a String method; I'd be fine with String returning the JSON representation.